### PR TITLE
update hasSBOM schema to include fields from ResourceDescriptor

### PR DESCRIPTION
**Pull request by pxp928**
Part of issue https://github.com/guacsec/guac/issues/754

- Update HasSBOM schema to include:
```
uri: String!
algorithm: String!
digest: String!
downloadLocation: String!
annotations: [Annotations!]!
```

- remove old annotation from backend implementation for HasSBOM
- implemented in memory resolver update for HasSBOM
- fix testdata error for `isDependency`

### CODE REVIEW

File: pkg/assembler/graphql/schema/hasSBOM.graphql [hunk 1]
```diff
@@ -30,38 +34,72 @@ type HasSBOM {
   id: ID!
   subject: PackageOrSource!
   uri: String!
-  annotation: String!
+  algorithm: String!
```
Comment (pxp928) [inline pkg/assembler/graphql/schema/hasSBOM.graphql:37]: @mihaimaruseac @mihaimaruseac wanted feedback if this meets all our needs before proceeding with the implementation.

File: pkg/assembler/backends/inmem/hasSBOM.go [hunk 2]
```diff
@@ -27,13 +27,12 @@ import (
...
-	uri        string
-	annotation string
-	origin     string
-	collector  string
+	id        uint32
+	pkg       uint32
+	src       uint32
+	uri       string
+	origin    string
+	collector string
```
Comment (mihaimaruseac) [inline pkg/assembler/backends/inmem/hasSBOM.go:35]: We're not storing the new fields?

Comment (pxp928) [reply pkg/assembler/backends/inmem/hasSBOM.go:35]: this was an old implementation where I added an `annotation` field. So I removed that for now and will create another PR to update the backends. I wanted to get feedback on the schema before moving forward.

Comment (pxp928) [reply pkg/assembler/backends/inmem/hasSBOM.go]: Just added to this PR.

File: cmd/guacgql/cmd/ingest.go [hunk 3]
```diff
@@ -1059,10 +1059,10 @@ func ingestHasSBOM(ctx context.Context, client graphql.Client) {
 			Tag:       &sourceTag,
 		},
 		hasSBOM: model.HasSBOMInputSpec{
-			Annotation: "this SBOM has a score of 10",
-			Uri:        "uri:location of source SBOM",
-			Origin:     "Demo ingestion",
-			Collector:  "Demo ingestion",
+			//Annotation: "this SBOM has a score of 10",
```
Comment (pxp928) [inline cmd/guacgql/cmd/ingest.go:1062]: Updating the test here

Comment (pxp928) [reply cmd/guacgql/cmd/ingest.go]: done.

**Comment by pxp928** [comment]
implemented in memory resolver update for HasSBOM

**Review by mlieberman85 — APPROVED** [review]
LGTM

**Pull request merged**